plugins/chain_plugin/include/eosio/chain_plugin/chain_plugin.hpp

```diff
@@ -942,6 +942,7 @@ class read_write : public api_base {
             // which is the format used by secondary index
             chain::key256_t k;
             uint8_t buffer[32];
+            memset(buffer, 0, 32);
             boost::multiprecision::export_bits(v, buffer, 8, false);
```

Comment:
Thanks for the fix, good catch! It would definitely work fine.
A thought, since `export_bits` returns the pointer to the byte just after the last one written, we can do a small optimization by zeroing only the remaining bytes. 

```suggestion
            auto out = boost::multiprecision::export_bits(v, buffer, 8, false);
            memset(out, 0, 32 - (out - buffer)); // zero out remaining bytes
```

Reply:
My hunch is the original is faster (although personally I'd prefer the `uint8_t buffer[32] = {}` style) because the call to `memset()` is completely removed (and there isn't a read dependency on the register).

https://godbolt.org/z/axGa6e4bK

Reply:
I have no issue with @spoonincode 's suggestion. Hard to know which version would be fastest without testing, but it probably doesn't matter in this instance.

Reply:
I'll add my vote for the more c++ style of `uint8_t buffer[32] = {}`